src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs

```diff
@@ -30,4 +32,20 @@ public override void ConfigureServices(IServiceCollection services)
 
         }
     }
+
+    [Feature("OrchardCore.ContentLocalization.ContentCulturePicker")]
+    public class ContentPickerStartup : StartupBase
+    {
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IContentCulturePickerService, ContentCulturePickerService>();
+            // todo: this does not work, find another way.
+            services.Configure<RequestLocalizationOptions>(options =>
+            {
+                options.RequestCultureProviders.Add(new ContentRequestCultureProvider()); 
```

Comment:
I would like feedback on where to add this `ContentRequestCultureProviders`

Reply:
I will take a look asap

Reply:
I think you just need to insert it before the `AcceptLanguageHeaderRequestCultureProvider` which is the last one in the list. Something like
```csharp
options.RequestCultureProviders.Insert(2, new ContentRequestCultureProvider());
```

Reply:
@Skrypt do you remember the first suggested **ordered** culture providers ;)

Reply:
It's the AcceptLanguageHeader one.

Reply:
If I add this, will this be applied to the admin as well ? We probably only want this on the frontend.

Reply:
Actually, It will probably be fine (admin will not be affected) since the url lookups we do are specific to the client applicaiton.